### PR TITLE
feat(conditional-writes): misc updates and fixes

### PR DESCRIPTION
**Pull request by david-leifker**
* Non-existent aspect validated against version `-1`
* Fix scroll sort order (added test)
* Documentation updates (batchGet, `-1` version)

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Summary by CodeRabbit

- **Documentation**
  - Updated advanced user and API usage guides with information on conditional writes and batch operations.
  - Added examples for fetching latest aspects, mutating global tags, and retrieving previous versions.

- **Refactor**
  - Transitioned internal data structures to use `LinkedHashMap` for better order preservation.
  - Updated method signatures across several controllers to reflect the new data structure.

- **Tests**
  - Enhanced test coverage for conditional writes and batch operations.
  - Added tests to ensure search result order preservation in entity controllers.

### CODE REVIEW

File: metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/controller/GenericEntitiesController.java [hunk 1]
```diff
@@ -128,13 +128,21 @@ protected List<E> buildEntityList(
                             aspectNames.stream()
                                 .map(aspectName -> Map.entry(aspectName, 0L))
                                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))))
-                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+                .collect(
+                    Collectors.toMap(
+                        Map.Entry::getKey,
+                        Map.Entry::getValue,
+                        (a, b) -> {
+                          throw new IllegalStateException("Duplicate key");
```
Comment (RyanHolstien) [inline metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/controller/GenericEntitiesController.java:136]: If it has the same value then it's likely okay, do we still want to throw an error for this case? Or can we put in a comment why the default map merger function is not appropriate?

File: metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/controller/GenericEntitiesController.java [hunk 2]
```diff
@@ -642,47 +652,54 @@ protected Boolean exists(
...
-                                  normalizedNames.get(reqEntry.getKey()), reqEntry.getValue()))
-                      .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
             })
-        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        .collect(
+            Collectors.toMap(
+                Map.Entry::getKey,
+                Map.Entry::getValue,
+                (a, b) -> {
+                  throw new IllegalStateException("Duplicate key");
```
Comment (RyanHolstien) [inline metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/controller/GenericEntitiesController.java:700]: Same here

**Review by RyanHolstien — APPROVED** [review]
One concern, but non-blocking

File: metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/controller/GenericEntitiesController.java (hunk 1 above)
Comment (david-leifker) [reply metadata-service/openapi-servlet/src/main/java/io/datahubproject/openapi/controller/GenericEntitiesController.java:136]: My understanding is that this is the default behavior of the map classes. Unfortunately the steaming interface doesn't allow deferring to the default behavior and forces me to re-implement the merge

**Pull request merged**